test/widgets/recurring_drive_detail_page_test.dart

```diff
@@ -0,0 +1,160 @@
+        until: DateTime.now().add(const Duration(days: 30)).toUtc(),
+      ),
+    );
+    whenRequest(processor).thenReturnJson(recurringDrive.toJsonForApi());
+  });
+
+  group('RecurringDriveDetailPage', () {
+    group('constructors', () {
+      testWidgets('Works with id parameter', (WidgetTester tester) async {
+        await pumpMaterial(tester, RecurringDriveDetailPage(id: recurringDrive.id!));
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+        // Wait for the recurring drive to be fully loaded
+        await tester.pump();
+
+        expect(find.byType(TripOverview), findsOneWidget);
+        expect(find.byType(WeekDayPicker), findsOneWidget);
+
+        expect(find.byType(DriveCard, skipOffstage: false),
+            findsNWidgets(RecurringDriveDetailPageState.shownDrivesCountDefault));
+      });
+
+      testWidgets('Works with object parameter', (WidgetTester tester) async {
+        await pumpMaterial(tester, RecurringDriveDetailPage.fromRecurringDrive(recurringDrive));
+
+        expect(find.byType(TripOverview), findsOneWidget);
+        expect(find.byType(WeekDayPicker), findsOneWidget);
+
+        expect(find.text(recurringDrive.start), findsOneWidget);
+
+        expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+        // Wait for the recurring drive to be fully loaded
+        await tester.pump();
+
+        expect(find.byType(DriveCard, skipOffstage: false),
+            findsNWidgets(RecurringDriveDetailPageState.shownDrivesCountDefault));
+      });
+    });
+
+    group('Upcoming drives', () {
+      final showLessButtonFinder = find.byKey(const Key('showLessButton'));
+      final showMoreButtonFinder = find.byKey(const Key('showMoreButton'));
+
+      Future<void> tapShowLessButton(WidgetTester tester, {double scrollDelta = 1}) async {
+        await tester.scrollUntilVisible(showLessButtonFinder, 50 * scrollDelta);
+        await tester.tap(showLessButtonFinder);
+        await tester.pump();
+      }
+
+      Future<void> tapShowMoreButton(WidgetTester tester, {double scrollDelta = 1}) async {
+        await tester.scrollUntilVisible(showMoreButtonFinder, 50 * scrollDelta);
+        await tester.tap(showMoreButtonFinder);
+        await tester.pump();
+      }
+
+      testWidgets('Shows upcoming drives', (WidgetTester tester) async {
+        await pumpMaterial(tester, RecurringDriveDetailPage.fromRecurringDrive(recurringDrive));
+        await tester.pump();
+
+        final driveCardFinder = find.byType(DriveCard, skipOffstage: false);
+        expect(driveCardFinder, findsNWidgets(RecurringDriveDetailPageState.shownDrivesCountDefault));
+        final driveCardState = tester.state<DriveCardState>(driveCardFinder.first);
+        expect(driveCardState.trip.id, equals(recurringDrive.drives!.first.id));
+      });
+
+      testWidgets('Can show less and more upcoming drives', (WidgetTester tester) async {
```

Comment:
this test failed, for me.

Reply:
It also seems to have failed in the build pipeline on the last commit, so that test is probably flaky. I'll look into it